packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -46,46 +48,31 @@ interface IInitializable {
 ///         expected to be set after a new deployment. Lastly, we reset the account code and storage slots the contract
 ///         was initially deployed to so it's not included in the `vm.dumpState`.
 contract L2Genesis is Script, Artifacts {
-    uint256 constant PROXY_COUNT = 2048;
-    uint256 constant PRECOMPILE_COUNT = 256;
     DeployConfig public constant cfg =
         DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
 
-    /// @notice The storage slot that holds the address of a proxy implementation.
-    /// @dev `bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)`
-    bytes32 internal constant PROXY_IMPLEMENTATION_ADDRESS =
-        0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
-
-    /// @notice The storage slot that holds the address of the owner.
-    /// @dev `bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)`
-    bytes32 internal constant PROXY_ADMIN_ADDRESS = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
-    uint80 internal constant DEV_ACCOUNT_FUND_AMT = 10_000 ether;
-    /// @notice Default Anvil dev accounts. Only funded if `cfg.fundDevAccounts == true`.
-    address[10] internal devAccounts = [
-        0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266,
-        0x70997970C51812dc3A010C7d01b50e0d17dc79C8,
-        0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC,
-        0x90F79bf6EB2c4f870365E785982E1f101E93b906,
-        0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65,
-        0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc,
-        0x976EA74026E726554dB657fA54763abd0C3a0aa9,
-        0x14dC79964da2C08b23698B3D3cc7Ca32193d9955,
-        0x23618e81E3f5cdF7f54C3d65f7FBc0aBf5B21E8f,
-        0xa0Ee7A142d267C1f36714E4a8F75612F20a79720
-    ];
-
     string internal outfile;
+    DeployConfig public deployConfig;
+
+    function getDeploymentContext() public view returns (string memory) {
+        return deploymentContext;
+    }
 
     /// @dev Reads the deploy config, sets `outfile` which is where the `vm.dumpState` will be saved to, and
     ///      loads in the addresses for the L1 contract deployments.
     function setUp() public override {
         Artifacts.setUp();
 
-        string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        string memory deployConfigPath = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        if (!vm.exists(deployConfigPath)) {
+            revert(string.concat("Deploy config not found at the expected path: ", deployConfigPath));
+        }
+
         vm.etch(address(cfg), vm.getDeployedCode("DeployConfig.s.sol:DeployConfig"));
         vm.label(address(cfg), "DeployConfig");
         vm.allowCheatcodes(address(cfg));
-        cfg.read(path);
+        cfg.read(deployConfigPath);
+        deployConfig = cfg;
```

Comment:
I am confused why we are setting `deployConfig` with `cfg` - seems like we are duplicating things

Reply:
It's to be able to export it as `public` from `L2Genesis.s.sol` so that `L2Genesis.t.sol` doesn't have to setup another deploy config, it can just read it from `L2Genesis.s.sol`